keel-sql/src/main/resources/db/changelog/20210226-generate-event-columns.yml

```diff
@@ -0,0 +1,80 @@
+databaseChangeLog:
+- changeSet:
+    id: generate-event-timestamp
+    author: fletch
+    changes:
+    - dropIndex:
+        tableName: event
+        indexName: event_uid_timestamp_idx
+    - dropIndex:
+        tableName: event
+        indexName: event_scope_ref_timestamp_idx
+    - dropIndex:
+        tableName: event
+        indexName: event_scope_application_timestamp_idx
+    - dropColumn:
+        tableName: event
+        columnName: ref
+    - dropColumn:
+        tableName: event
+        columnName: application
+    - dropColumn:
+        tableName: event
+        columnName: timestamp
+    - sql:
+        sql: |
+          alter table event
+          add column ref varchar(255) generated always as (json ->> '$.ref')
+          not null
+          after scope;
+    - sql:
+        sql: |
+          alter table event
+          add column application varchar(255) generated always as (json ->> '$.application')
+          not null
+          after ref;
```

Comment:
`ref` and `application` are strings pulled from the JSON. Actually `application` already was but it was in a non-optimal position in the table, so I thought it made sense to move it.